content/posts/how-to-evolve-application-cicd/index.md

```diff
@@ -0,0 +1,336 @@
+            - v1-dependencies-{{ checksum "build.gradle" }}
+            # fallback to using the latest cache if no exact match is found
+            - v1-dependencies-
+
+      - run: ./gradlew assemble
+
+      - save_cache:
+          paths:
+            - ~/.gradle
+          key: v1-dependencies-{{ checksum "build.gradle" }}
+      - persist_to_workspace:
+          root: ~/happraisal
+          paths:
+            - ./build
+      - store_artifacts:
+          path: ~/happraisal/build/libs
+
+unit-tests:
+docker:
+- image: cimg/openjdk:17.0.5
+
+    working_directory: ~/happraisal
+    steps:
+        - checkout
+        - restore_cache:
+            keys:
+              - v1-dependencies-{{ checksum "build.gradle" }}
+              # fallback to using the latest cache if no exact match is found
+              - v1-dependencies-
+        - run: ./gradlew test
+        - save_cache:
+            paths:
+              - ~/.gradle
+            key: v1-dependencies-{{ checksum "build.gradle" }}
+        - store_test_results:
+            path:  ~/happraisal/build/test-results
+
+
+
+integration-tests:
+docker:
+- image: cimg/openjdk:17.0.5
+
+    working_directory: ~/happraisal
+    steps:
+        - checkout
+        - restore_cache:
+            keys:
+              - v1-dependencies-{{ checksum "build.gradle" }}
+              # fallback to using the latest cache if no exact match is found
+              - v1-dependencies-
+        - run: ./gradlew integrationTest
+        - save_cache:
+            paths:
+              - ~/.gradle
+            key: v1-dependencies-{{ checksum "build.gradle" }}
+
+
+system-tests:
+docker:
+- image: cimg/openjdk:17.0.5
+
+    working_directory: ~/happraisal
+    steps:
+        - checkout
+        - restore_cache:
+            keys:
+              - v1-dependencies-{{ checksum "build.gradle" }}
+              # fallback to using the latest cache if no exact match is found
+              - v1-dependencies-
+        - run: ./gradlew systemTest
+        - save_cache:
+            paths:
+              - ~/.gradle
+            key: v1-dependencies-{{ checksum "build.gradle" }}
+
+workflows:
+version: 2.1
+microservice:
+jobs:
+- assemble
+- unit-tests:
+requires:
+- assemble
+- integration-tests:
+requires:
+- assemble
+- system-tests:
+requires:
+- assemble
+
+````
+
+
+
+
+
+### Implementation with GitHub Action
+
+#### Terminology
+
+Workflow: In the official documentation a workflow is defined as: "A workflow is a configurable automated process that will execute one or more jobs.
+It is set up via a yaml file
+https://docs.github.com/fr/actions/using-workflows/about-workflows
+
+action: "Actions are individual tasks that you can combine to create jobs and customize your workflow."
+https://docs.github.com/fr/actions/creating-actions/about-custom-actions
+
+####  first connection
+
+GitHub Action is the integrated continuous integration tool of GitHUb.
+It is accessible from the Action button in a toolbar
```

Comment:
What is this "it"? Is it the integration tool? GitHub?

Reply:
The integration tool